### PR TITLE
2588/radius robin

### PR DESCRIPTION
**Pull request by maxbeth**
For better redundancy, the radius can now round-robin.
Issue #2588

### CODE REVIEW

File: privacyidea/api/radiusserver.py [hunk 1]
```diff
@@ -127,9 +127,16 @@ def test():
     password = getParam(param, "password", required)
     dictionary = getParam(param, "dictionary",
                           default="/etc/privacyidea/dictionary")
-
-    r = test_radius(identifier, server, secret, user, password, port=port,
+    r = 0
+    for server in servers.split(','):
```
Comment (cornelinux) [inline privacyidea/api/radiusserver.py:131]: Please do

    for server in [x.strip() for x in servers.split(','):

This way we will also be able to handle lists like

1.2.3.4:1812, 2.3.4.5:1645

which contain whitespaces.

File: privacyidea/api/radiusserver.py [hunk 2]
```diff
@@ -127,9 +127,16 @@ def test():
...
+        if len(serveraddr) >= 2:
+            serverport = int(serveraddr[1])
+        else:
+            serverport = port
+        r = test_radius(identifier, serveraddr[0], secret, user, password, port=serverport,
                     dictionary=dictionary, retries=retries, timeout=timeout)
-    g.audit_object.log({'success': r > 0,
+        g.audit_object.log({'success': r > 0,
                         'info':  r})
     return send_result(r > 0)
```
Comment (cornelinux) [inline privacyidea/api/radiusserver.py:142]: Hm, I am not sure, what the result should be here.

So if the first server fails, but the 2nd succeeds, we get a success.
If the first is successful and the 2nd fails, we get a fail.

**Review by cornelinux — CHANGES_REQUESTED** [review]
I have not checked the radiustoken.py, yet.
But please see my comments in radiusserver.py.
Also add some docs about redundancy to 
https://github.com/privacyidea/privacyidea/blob/master/doc/configuration/radiusserver_config.rst

File: privacyidea/api/radiusserver.py (hunk 1 above)
Comment (maxbeth) [reply privacyidea/api/radiusserver.py:131]: In the privacyidea//lib/tokens/radiustoken.py is a strip also

File: privacyidea/api/radiusserver.py (hunk 2 above)
Comment (maxbeth) [reply privacyidea/api/radiusserver.py:144]: I return now the first response

**Comment by cornelinux** [comment]
THanks a lot for the input.
We are thinking about reworking the whole radius implementation.
So I am closing this for now.

**Pull request closed without merging**